src/Elsa.Dashboard/Pages/_DataDictionaryRecord.cshtml.cs

```diff
@@ -56,14 +57,13 @@ public async Task<IActionResult> OnPostAsync()
         await _client.UpdateDataDictionaryRecord(_serverUrl, DictionaryRecord);
       }
       else { HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest; }
-      return RedirectToPage("DataDictionaryRecord", new {group = GroupId, record = DictionaryRecord!.Id });
+      return await OnGetAsync(GroupId, DictionaryRecord!.Id);
     }
 
     public async Task<IActionResult> OnPostArchiveAsync(bool archive)
     {
       await _client.ArchiveDataDictionaryRecord(_serverUrl, RecordId, archive);
-      return RedirectToPage("DataDictionaryRecord", new { group = GroupId, record = DictionaryRecord!.Id });
-
+      return await OnGetAsync(GroupId, RecordId);
```

Comment:
This caused Some issues with the URL last time I tried this - hence the need for using the Redirect. I think the issue (fixed in a different pull request was the page name was incorrect -"_DataDictionaryRecord"